packages/xen-api/src/_parseUrl.js

```diff
@@ -1,13 +1,21 @@
-const URL_RE = /^(?:(https?:)\/*)?(?:([^:]+):([^@]+)@)?(?:\[([^\]]+)\]|([^:/]+))(?::([0-9]+))?(\/[^?#]*)?$/
+const URL_RE = /^(?:(https?:)\/*)?(?:(([^:]+):([^@]+))@)?(?:\[([^\]]+)\]|([^:/]+))(?::([0-9]+))?(\/[^?#]*)?$/
```

Comment:
I'm not sure it match a token without `:` , is it ok ? 
also there is a unescaped  / in `([^:/]+)`

Reply:
1. Good catch, I'll fix this :clap: 
2. Indeed, it should not be escaped in character classes.

Reply:
1. Fixed and tests added.

Reply:
1. Fixed and tests added.